src/pages/StocktakeManagePage.js

```diff
@@ -1,290 +1,201 @@
+import { buttonStrings, modalStrings } from '../localization';
+import { UIDatabase } from '../database';
+import { BottomTextEditor } from '../widgets/modals';
+import { ToggleBar } from '../widgets';
+import { DataTable, DataTableHeaderRow, DataTableRow } from '../widgets/DataTable';
+import {
+  sortData,
+  filterData,
+  selectRow,
+  deselectRow,
+  selectAll,
+  deselectAll,
+  hideStockOut,
+  showStockOut,
+  selectItems,
+  editName,
+} from './dataTableUtilities/actions';
+
+import globalStyles, { SUSSOL_ORANGE, newDataTableStyles, newPageStyles } from '../globalStyles';
+import usePageReducer from '../hooks/usePageReducer';
+import DataTablePageView from './containers/DataTablePageView';
+import { createStocktake, addItemsToStocktake } from '../navigation/actions';
+import { recordKeyExtractor, getItemLayout } from './dataTableUtilities/utilities';
+
+export const StocktakeManagePage = ({
+  routeName,
+  dispatch: reduxDispatch,
+  stocktake,
+  runWithLoadingIndicator,
+}) => {
+  const [state, dispatch, instantDebouncedDispatch, debouncedDispatch] = usePageReducer(routeName, {
+    pageObject: stocktake,
+    backingData: UIDatabase.objects('Item'),
+    data: UIDatabase.objects('Item')
+      .sorted('name')
+      .slice(),
+    keyExtractor: recordKeyExtractor,
+    dataState: new Map(),
+    searchTerm: '',
+    filterDataKeys: ['name', 'code'],
+    sortBy: 'name',
+    isAscending: true,
+    hasSelection: false,
+    allSelected: false,
+    showAll: true,
+    name: stocktake ? stocktake.name : '',
+  });
+
+  const {
+    data,
+    dataState,
+    sortBy,
+    isAscending,
+    columns,
+    hasSelection,
+    showAll,
+    allSelected,
+    name,
+    keyExtractor,
+  } = state;
+
+  // On navigating to this screen, if a stocktake is passed through, update the selection with
+  // the items already in the stocktake.
+  useEffect(() => {
+    if (stocktake) dispatch(selectItems(stocktake.itemsInStocktake));
+  }, []);
+
+  const getAction = useCallback((colKey, propName) => {
+    switch (colKey) {
+      case 'selected':
+        if (propName === 'onCheckAction') return selectRow;
+        return deselectRow;
+      default:
+        return null;
     }
-
-    if (stocktake) {
-      const selected = [];
-      stocktake.items.forEach(stocktakeItem => {
-        const { itemId } = stocktakeItem;
-        if (itemId !== '') selected.push(itemId);
-      });
-
-      this.setState(
+  });
+
+  const Toggle = () => (
+    <ToggleBar
+      style={globalStyles.toggleBar}
+      textOffStyle={globalStyles.toggleText}
+      textOnStyle={globalStyles.toggleTextSelected}
+      toggleOffStyle={globalStyles.toggleOption}
+      toggleOnStyle={globalStyles.toggleOptionSelected}
+      toggles={[
         {
-          selection: selected,
-          stocktakeName: stocktake.name,
+          text: buttonStrings.hide_stockouts,
+          onPress: () => (showAll ? dispatch(hideStockOut()) : dispatch(showStockOut())),
```

Comment:
this is confusing - should dispatch a `toggleStockOut` and let the action creator determine which action to return to dispatch

Reply:
yep